packages/utils/network.ts

```diff
@@ -1,15 +1,37 @@
 export const fetchWithTimeout = async (
   timeout: number,
   ...params: Parameters<typeof fetch>
-) => {
-  const controller = new AbortController()
-  const timeoutId = setTimeout(() => controller.abort(), timeout)
-  try {
-    return await fetch(params[0], {
-      ...params[1],
-      signal: controller.signal,
-    })
-  } finally {
-    clearTimeout(timeoutId)
-  }
-}
+) =>
+  new Promise<Response>(async (resolve, reject) => {
+    const controller = new AbortController()
+
+    let rejected = false
+    const timeoutId = setTimeout(() => {
+      console.log('Aborting...')
```

Comment:
Aborting what?

Reply:
This aborts the fetch request below after 3 seconds. Added more info to the log and comments 👍